pkg/controller/elasticsearch/client/base.go

```diff
@@ -133,6 +134,9 @@ func (c *baseClient) request(
 		return err
 	}
 
+	// Sets headers allowing ES to distinguish between deprecated APIs used internally and by the user
+	common.SetInternalProductRequestHeader(request)
```

Comment:
nit: I would update the unit test [TestClient_request](https://github.com/elastic/cloud-on-k8s/blob/b3f4ae2fd7b281f5a9a8693132772b4a347d409f/pkg/controller/elasticsearch/client/client_test.go#L236) to ensure that the header is set.